moneriote/moneriote.py

```diff
@@ -0,0 +1,234 @@
+            elif isinstance(output, str):
+                data['md_height'] = int(re.sub('[^0-9]', '', output.splitlines()[0]))
+                log_msg('monerod height is %d' % data['md_height'])
+                if method == 'monerod':
+                    return data['md_height']
+
+        if method in ['compare', 'moneroblocks']:
+            retries = 0
+            while True:
+                if retries > max_retries:
+                    break
+                try:
+                    blob = make_json_request('https://moneroblocks.info/api/get_stats/', timeout=5, verify=True)
+                    data['moneroblocks'] = blob.get('height')
+                    break
+                except Exception as ex:
+                    log_msg('Fetching moneroblocks JSON has failed. Retrying.')
+                    retries += 1
+                    time.sleep(1)
+
+        if method in ['compare', 'xmrchain']:
+            retries = 0
+            while True:
+                if retries > max_retries:
+                    break
+                try:
+                    blob = make_json_request('https://xmrchain.net/api/networkinfo', timeout=5, verify=True)
+                    assert blob.get('status') == 'success'
+                    data['xmrchain_height'] = blob.get('data', {}).get('height')
+                    assert isinstance(data['xmrchain_height'], int)
+                    log_msg('xmrchain height is %d' % data['xmrchain_height'])
+                    if method == 'xmrchain':
+                        return data['xmrchain_height']
+                    break
+                except Exception as ex:
+                    log_msg('Fetching xmrchain JSON has failed. Retrying.')
+                    retries += 1
+                    time.sleep(1)
+                    continue
+
+        if data:
+            return max(data.values())
+        log_err('Unable to obtain blockheight.')
+
+    def monerod_get_peers(self):
+        """Gets the last known peers from monerod"""
+        nodes = RpcNodeList()
+        output = self._daemon_command("print_pl")
+        if not output:
+            return nodes
+
+        regex = r"(gray|white)\s+(\w+)\s+(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}):(\d{1,5})"
+        matches = re.finditer(regex, output)
+
+        for i, match in enumerate(matches):
+            if match.group(1) != 'white':
+                continue
+
+            address = match.group(3)
+            nodes.append(RpcNode(address=address))
+
+        log_msg('Got peers from RPC: %d node(s)' % len(nodes))
+        return nodes
+
+    def _daemon_command(self, cmd: str):
+        if not os.path.exists(self.md_path):
+            log_err("monerod not found in path \'%s\'" % self.md_path)
+            return
+
+        log_msg("Spawning daemon; executing command \'%s\'" % cmd)
+
+        # build proc args
+        args = [
+            '--rpc-bind-ip', self.md_daemon_addr,
+            '--rpc-bind-port', str(18081),
```

Comment:
@skftn , I finally found reason here, my port setting in config.ini is 18083. I tried str(18083) and every thing works!

Reply:
You are correct. Sorry, I will fix :)